staging/src/k8s.io/apimachinery/pkg/apimachinery/announced/group_factory.go

```diff
@@ -89,6 +89,17 @@ func (gmf *GroupMetaFactory) Announce(groupFactoryRegistry APIGroupFactoryRegist
 		panic(err)
 	}
 	return gmf
+
+}
+
+// AnnounceIfNotAnnounced is the same as Announce, except that it returns nil
```

Comment:
If Announce suffers from a race/double import, it's probably better to fix it in Announce rather than add a new way to do things. If not, Announce needs a warning about not calling it twice.